src/profiling/meter_manager.cpp

```diff
@@ -1,5 +1,5 @@
+#include <communication/distributed_context.hpp>
```

Comment:
Here may be en example where the templating should stay at the current level, but the communicator itself should have a handle that erases the templating.

Reply:
I don't quite follow. Which templates?
This refactoring removes the template from the `communicator` and passes the communicator a handle.
The handle has to perform virtual dispatch at some point: i.e. it has to hold onto pointers to the implementation of the various API methods.
You can do this with an abstract base class (i.e. typical OO C++), or with the type erasure method we use, or with a roll your own vtable.
The `distributed_context` type is the wrapper/handle type. It holds onto the implementation (`mpi_context` or `local_context`), and redirects user calls to their implementation in the "held" implementation.